apps/web/pages/auth/setup/index.tsx

```diff
@@ -9,26 +10,83 @@ import prisma from "@calcom/prisma";
 import { inferSSRProps } from "@calcom/types/inferSSRProps";
 import { WizardForm } from "@calcom/ui";
 
-import SetupFormStep1 from "./steps/SetupFormStep1";
+import AdminUser from "./steps/AdminUser";
+import ChooseLicense from "./steps/ChooseLicense";
+import EnterpriseLicense from "./steps/EnterpriseLicense";
 import StepDone from "./steps/StepDone";
 
 export default function Setup(props: inferSSRProps<typeof getServerSideProps>) {
   const { t } = useLocale();
-  const [isLoadingStep1, setIsLoadingStep1] = useState(false);
-  const shouldDisable = props.userCount !== 0;
+  const [isLoadingAdminUser, setIsLoadingAdminUser] = useState(false);
+  const [isFreeLicense, setIsFreeLicense] = useState(
+    props.deployment?.licenseType !== DeploymentLicenseType.EE
+  );
+  const [isLoadingAdminApps, setIsLoadingAdminApp] = useState(false);
+  const [isLoadingChooseLicense, setIsLoadingChooseLicense] = useState(false);
+  const [isLoadingEnterpriseLicense, setIsLoadingEnterpriseLicense] = useState(false);
+  const [isEnabledEnterpriseLicense, setIsEnabledEnterpriseLicense] = useState(
+    !!props.deployment?.licenseKey
+  );
```

Comment:
These are states to keep track of various things among steps.

Reply:
That's a lot of useStates

Reply:
Tracking some of these in rhf or even into a single state object would probably be the best here 

From what i can see - a lot of these are controlled by what are effectively radio buttons and would work directly in rhf

Reply:
Refactored to make the wizard handle it by the step content.